axum-core/src/response/mod.rs

```diff
@@ -149,38 +144,17 @@ pub trait IntoResponse {
 }
 
 /// Trait for generating response headers.
-///
-
-/// **Note: If you see this trait not being implemented in an error message, you are almost
-/// certainly being mislead by the compiler¹. Look for the following snippet in the output and
-/// check [`IntoResponse`]'s documentation if you find it:**
-///
-/// ```text
-/// note: required because of the requirements on the impl of `IntoResponse` for `<type>`
-/// ```
-///
-/// Any type that implements this trait automatically implements `IntoResponse` as well, but can
-/// also be used in a tuple like `(StatusCode, Self)`, `(Self, impl IntoResponseHeaders)`,
-/// `(StatusCode, Self, impl IntoResponseHeaders, impl IntoResponse)` and so on.
-///
-/// This trait can't currently be implemented outside of axum.
```

Comment:
The current trait members are rather weird, do you really want them to be part of the public API nominally?

Reply:
Yeah they're quite odd at first glance. I suppose we could make them `#[doc(hidden)]` 🤔

Reply:
I think we should.